src/lib/legacy.js

```diff
@@ -697,6 +697,39 @@ export function getBuyGlpFromAmount(toAmount, fromTokenAddress, infoTokens, glpP
   return { amount: fromAmount, feeBasisPoints };
 }
 
+export function getLowestFeeTokenForBuyGlp(
+  chainId,
+  toAmount,
+  glpPrice,
+  usdgSupply,
+  totalTokenWeights,
+  infoTokens,
+  fromTokenAddress
+) {
+  if (!chainId || !toAmount || !infoTokens || !glpPrice || !usdgSupply || !totalTokenWeights) {
+    return;
+  }
+  const tokens = getVisibleTokens(chainId);
+  const usdgAmount = toAmount.mul(glpPrice).div(PRECISION);
+  return tokens
+    .map((token) => {
+      const fromToken = getTokenInfo(infoTokens, token.address);
+      const fees = getFeeBasisPoints(
+        fromToken,
+        usdgAmount,
+        MINT_BURN_FEE_BASIS_POINTS,
+        TAX_BASIS_POINTS,
+        true,
+        usdgSupply,
+        totalTokenWeights
+      );
+      return { ...token, fees };
+    })
+    .filter((t) => t.address !== fromTokenAddress)
+    .filter((t) => t.fees)
```

Comment:
let's change it to single filter call as well?

Reply:
Done!